pkg/job-runtime-analyzer/job-runtime-analyzer.go

```diff
@@ -0,0 +1,202 @@
+package jobruntimeanalyzer
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kataras/tablewriter"
+	"github.com/montanaflynn/stats"
+	corev1 "k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func getCacheBasedir() string {
+	if runtime.GOOS == "linux" {
+		// Survives reboots and is not ram-backed
+		return "/var/tmp/job-runtime-analyzer"
+	}
+	return "/tmp/job-runtime-analyzer"
+}
+
+type podContainerRuntime struct {
+	podName       string
+	containerName string
+	finished      bool
+	duration      time.Duration
+}
+
+func Run(podJSONURL string) error {
+	rawPodList, err := fetchPodJSON(podJSONURL)
+	if err != nil {
+		return fmt.Errorf("failed to fetch pod json: %w", err)
+	}
+
+	podList := &corev1.PodList{}
+	if err := json.Unmarshal(rawPodList, podList); err != nil {
+		return fmt.Errorf("failed to unmarshal pod list: %w", err)
+	}
+
+	var runtimes []*podContainerRuntime
+	for _, pod := range podList.Items {
+		for _, container := range append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...) {
+			if pod.Name == "test" {
```

Comment:
This check shouldn't be in this loop.

Reply:
The check is just wrong, should be `container.Name`